controllers/inverse_kinematics_pyikfast/inverse_kinematics_pyikfast.py

```diff
@@ -80,27 +80,74 @@
             sensors.append(sensor)
             sensor.enable(timeStep)
         except Exception as e:
-            print('Rotational Motor: ' + device.getName() +
-                  ' has no Position Sensor')
+            print("Rotational Motor: " + device.getName() + " has no Position Sensor")
+print("minPositions", minPositions)
+print("maxPositions", maxPositions)
+
 # --------------------------------------------------------------------
+# move the arm onto target using known solution
+target_position_joint_state = np.array([0, -np.pi / 2, np.pi / 2, 0, 0, 0]).astype(
+    np.double
+)
+def move_to_joint_state(motors, joint_state):
+    """Move robot endpoint to a known position that aligns it perfectly with the target"""
+    for i in range(len(joint_state)):
+        motors[i].setPosition(joint_state[i])
+# plase end-effector on target
+move_to_joint_state(motors, target_position_joint_state)
 
+# --------------------------------------------------------------------
+# show that the endpoint pose is indeed the targtet pose
 # Initialize our inverse kinematics module
 ik = inverseKinematics(pyikfast_module, protoName, minPositions, maxPositions)
+print("ik", ik)
+
+fk_pos, fk_rot = ik.get_fk(target_position_joint_state)
+print("fk_pos", fk_pos)
+print("fk_rot", fk_rot)
+
+# check that the target position and rotation matches te end-effector's
+target = supervisor.getFromDef("TARGET")
+# target position relative to world.
+target_pos_world = np.array(target.getPosition())
+# target rotation relative to world
+target_rot_world = np.array(target.getOrientation()).reshape(3, 3)
+assert np.allclose(fk_pos, target_pos_world)
+assert np.allclose(fk_rot, target_rot_world)
+print("Yay, target pose and end-effector pose match")
+
+# --------------------------------------------------------------------
+# Try is get an IK solution for the current end-point pose.
+# there should be at least one solution, target_position_joint_state
```

Comment:
what is the purpose of this? Does not seem necessary for the initialization phase of the controller. Looks like troubleshooting / testcase code. I suggest creating a new troubleshooting-controller. The main controller should be as bare-bones and generic as possible.